src/components/course/CourseSkills.jsx

```diff
@@ -1,30 +1,33 @@
 import React, { useContext, useState } from 'react';
 import { Badge, Button } from '@edx/paragon';
 
+import { AppContext } from '@edx/frontend-platform/react';
 import { CourseContext } from './CourseContextProvider';
 import { SKILLS_BUTTON_LABEL } from './data/constants';
 
 export const MAX_VISIBLE_SKILLS = 5;
 
 export default function CourseSkills() {
+  const { enterpriseConfig } = useContext(AppContext);
   const { state } = useContext(CourseContext);
   const { skillNames } = state.course;
   const [showMore, setShowMore] = useState(false);
   const skillsButtonLabel = showMore ? SKILLS_BUTTON_LABEL.SHOW_LESS : SKILLS_BUTTON_LABEL.SHOW_MORE;
-
   return (
     <div className="mb-5">
       <h5> Skills you&apos;ll gain</h5>
       <div>
         {skillNames.map((skill, index) => (
-          <Badge
-            key={skill.id}
-            className="course-skill"
-            variant="light"
-            style={{ display: ((index < MAX_VISIBLE_SKILLS) || showMore) ? 'inline-block' : 'none' }}
-          >
-            { skill }
-          </Badge>
+          <a href={`/${enterpriseConfig.slug}/search?skill_names=${skill}`}>
```

Comment:
We have a [Hyperlink](https://edx.github.io/paragon/components/hyperlink/) component in paragon, so what if we use that but style it as badge? As per my understanding things will be simpler this way.

Reply:
Wouldn't it make this more complex and since we have a component available then shouldn't we use it except for writing a similar CSS to that?

Reply:
@zamanafzal  Just had a discussion with fedx team and there is another simple thing possible. Please see https://edx-internal.slack.com/archives/CD4E4BCM9/p1618234357340500

Reply:
@muhammad-ammar Great suggestion. Thank you for the heads up. 💯

Reply:
An additional suggestion might be to use react-router's `Link` component instead of the native anchor tag (i.e., `<a>`). `Link` takes advantage of React's routing to navigate to a page route that exists in the same MFE application, avoiding a full page refresh. This will result in better perceived performance for the user.

```jsx
import { Link } from 'react-router-dom';

<Badge
  variant="light"
  as={Link}
  to={`/${enterpriseConfig.slug}/search?skill_names=${skill}`}
>
  {skill}
</Badge>

Reply:
By doing this change, test cases are failing with the following error as this file is being called in the header and I think we need to change the structure of calling the header in Router to fix this case.
` Invariant failed: You should not use <Link> outside a <Router>`
Can we move forward with <a> tag for now and create a ticket to fix this issue.

Reply:
@zamanafzal There is a test utility for rendering a component nested within a router. Example:

```jsx
import { renderWithRouter } from '../../../utils/tests';

renderWithRouter(
  <CourseSkillsWithContext
    initialAppState={initialAppState}
    initialCourseState={courseState}
  />,
);
```

Reply:
Thank you @adamstankiewicz for helping on this. :)